cores/arduino/HardwareTimer.h

```diff
@@ -118,6 +118,9 @@ class HardwareTimer {
     void setMode(uint32_t channel, TimerModes_t mode, PinName pin = NC); // Configure timer channel with specified mode on specified pin if available
     void setMode(uint32_t channel, TimerModes_t mode, uint32_t pin);
 
+    void setPreloadEnable(bool value);
```

Comment:
I let @ABOSTM review this but just one question:
Why not declare only one function:

```suggestion
    void setPreloadEnable(bool value, uint32_t channel = 0);
```

Then doing the right stuff depending of the channel.

Reply:
I just followed the example of `attachInterrupt`, which also has two versions and puts the channel argument first. Also, both versions have no code in common, so merging these would not really simplify things (on the contrary, with separate functions things code is more clearly separated, I think).

If you still think changing this is a good idea, I'll do that, but I'm not so convinced it is actually better :-)

Reply:
ok ;)that's fine for me.